### PR TITLE
Fix #6951 Index.vue and Error404.vue for Typescript projects

### PR DESCRIPTION
**Pull request by rfox12**
Vue.component() or Vue.extend() is needed when script lang="ts"
https://vuejs.org/v2/guide/typescript.html#Basic-Usage
This will close issue #6951 https://github.com/quasarframework/quasar/issues/6951

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch and _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] It's been tested with all Quasar themes
- [ ] It's been tested on a Cordova (iOS, Android) app
- [ ] It's been tested on a Electron app
- [ ] Any necessary documentation has been added or updated [in the docs](https://github.com/quasarframework/quasar/tree/dev/docs) (for faster update click on "Suggest an edit on GitHub" at bottom of page) or explained in the PR's description.

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by IlCallo** [comment]
Seems good, but can you do the same for `layouts/MainLayout.vue`, `pages/Error404.vue` and `components/EssentialLink.vue`?

**Comment by rfox12** [comment]
MainLayout.vue and EssentialLink.vue are standard JS and run without error.  I suggest merge and consider additional changes later.  This PR will prevent new users from having a frustrating start with Quasar.

**Comment by rstoenescu** [comment]
Thanks for contributing!

File: template/src/pages/Error404.vue [hunk 1]
```diff
@@ -19,7 +19,15 @@
 </template>
 
 <script {{#preset.typescript}}lang="ts"{{/preset.typescript}}>
+{{#if preset.typescript}}
+import Vue as 'vue';
```
Comment (rstoenescu) [inline template/src/pages/Error404.vue:23]: import Vue from 'vue'

**Pull request merged**